sha3_api.c

```diff
@@ -15,6 +16,12 @@ void sha3_init(sha3_var_t *c, size_t md_sz)
   size_t i;
 
   for (i = 0; i < 25; i++)
+  __loop__(
+    assigns(i, object_whole(c->st.d))
+    invariant(i <= 25)
+    invariant(forall(j, 0, i, c->st.d[j] == 0))
+    decreases(25 - i)
```

Comment:
We usually leave out decreases. decreases is only needed to prove that this loop indeed terminates.
Termination is not usually a problem in cryptography (and where it is - e.g., when using unbounded rejection sampling - CBMC won't help anyway), so let's leave it out for now.